### PR TITLE
Implement ListDomain API

### PR DESCRIPTION
**Pull request by longquanzheng**

### CODE REVIEW

File: service/frontend/workflowHandler.go [hunk 1]
```diff
@@ -268,6 +268,50 @@ func (wh *WorkflowHandler) RegisterDomain(ctx context.Context, registerRequest *
...
+	listRequest *gen.ListDomainRequest) (*gen.ListDomainResponse, error) {
+	scope := metrics.FrontendListDomainScope
+	sw := wh.startRequestProfile(scope)
+	defer sw.Stop()
+
+	if listRequest == nil {
+		return nil, wh.error(errRequestNotSet, scope)
+	}
+
+	pageSize := 100
```
Comment (wxing1292) [inline service/frontend/workflowHandler.go]: care to make this configurable? in the service.go file

File: idl/github.com/uber/cadence/shared.thrift [hunk 2]
```diff
@@ -802,8 +802,18 @@ struct RegisterDomainRequest {
   80: optional map<string,string> data
 }
 
+struct ListDomainRequest {
+  10: optional i32 pageSize
+  20: optional string nextPageToken
```
Comment (samarabbas) [inline idl/github.com/uber/cadence/shared.thrift]: Can you binary blob for nextPageToken.

File: service/frontend/workflowHandler.go (hunk 1 above)
Comment (longquanzheng) [reply service/frontend/workflowHandler.go]: I would suggest not to use config file for this number unless you really need it.
1. User can always pass in a pageSize in the API. That would be the most use cases.
2. It is not necessary that we have to use different default page size in different environment(staging/sjc/dca/autobots/public repo/etc)
3. Having config will add more code and more configs, which would introduce lots of unnecessary code/config(staging/sjc/dca/autobots/public repo/etc). 
4. If having it in config, we still need to verify if the config value is set or not. If not, do you want to exit the process? Or use a another default value from code? Both look unnecessary. 

I am trying to understand why you want this. If this hardcoding number bothers you, I think we can use a constant number defined on the top of the file or somewhere else.

File: idl/github.com/uber/cadence/shared.thrift (hunk 2 above)
Comment (longquanzheng) [reply idl/github.com/uber/cadence/shared.thrift]: Done

File: service/frontend/workflowHandler.go (hunk 1 above)
Comment (wxing1292) [reply service/frontend/workflowHandler.go]: if you take a look at the service.go file or at least the top of the file you are changing, you can see a lot of const, which makes it easy to manage.

defining a arbitrary number everywhere in the code is not a good practice.

**Pull request merged**